frontend/src/views/Storage/Storage.tsx

```diff
@@ -10,66 +10,86 @@ import {
-import { DEFAULT_NMPFILE_YEAR, DefaultManureFormData } from '../../constants';
 import { StyledContent } from './storage.styles';
-import useAppState from '../../hooks/useAppState';
 import { NUTRIENT_ANALYSIS, MANURE_IMPORTS } from '../../constants/routes';
 
 import { AppTitle, PageTitle, ProgressStepper, TabsMaterial } from '../../components/common';
 import { addRecordGroupStyle, customTableStyle, tableActionButtonCss } from '../../common.styles';
 import StorageModal from './StorageModal';
+import useAppState from '@/hooks/useAppState';
+import { NMPFileManureStorageSystemsData } from '@/types';
+import DefaultNMPFileManureStorageSystemsData from '@/constants/DefaultNMPFileManureStorage';
 
 export default function Storage() {
-  const { state } = useAppState();
+  const { state, dispatch } = useAppState();
   const navigate = useNavigate();
-
-  // TODO: make correct file type
-  const [storageList, setStorageList] = useState<NMPFileFarmManureData[]>(
-    state.nmpFile.years[0]?.FarmManures || [],
+  const [rowEditIndex, setRowEditIndex] = useState<number | undefined>(undefined);
+  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
+  const [storageList, setStorageList] = useState<NMPFileManureStorageSystemsData[]>(
+    state.nmpFile.years[0]?.ManureStorageSystems || [],
+  );
+  const [storageFormData, setStorageFormData] = useState<NMPFileManureStorageSystemsData>(
+    DefaultNMPFileManureStorageSystemsData,
   );
-  // Not sure if NMPFileFarmManureData is the right file type
-  const [storageFormData, setstorageFormData] = useState<NMPFileFarmManureData>({
-    ManureSource: '',
-    MaterialType: '',
-    BookLab: '',
-    UniqueMaterialName: '',
-    Nutrients: { N: 0, P2O5: 0, K2O: 0, Moisture: '', NH4N: 0 },
-  });
-
-  const handleSubmit = () => {};
 
   const handlePrevious = () => {
     navigate(MANURE_IMPORTS);
   };
 
+  // save to NMPFileManureStorageSystemsData
+  // either imported or generated manure do we save IsMaterialStored and AssignedToStoredSystem booleans
   const handleNext = () => {
+    dispatch({
+      type: 'SAVE_MANURE_STORAGE_SYSTEMS',
+      year: state.nmpFile.farmDetails.Year!,
+      newManureStorageSystems: storageList,
+    });
     navigate(NUTRIENT_ANALYSIS);
   };
 
-  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
-
   const handleDialogClose = () => {
+    setRowEditIndex(undefined);
+    setStorageFormData(DefaultNMPFileManureStorageSystemsData);
     setIsDialogOpen(false);
   };
 
+  // on edit text fields dont show up
   const handleEditRow = (e: GridRenderCellParams) => {
+    setRowEditIndex(e.row);
+    setStorageFormData(e.row);
     setIsDialogOpen(true);
   };
 
-  const handleDeleteRow = (e: GridRenderCellParams) => {};
+  const handleDeleteRow = (e: GridRenderCellParams) => {
+    setStorageList((prev) => prev.filter((ele) => ele.Name !== e.row.Name));
+  };
+
+  const handleSubmit = (formData: NMPFileManureStorageSystemsData) => {
```

Comment:
Similarly, the `handleSubmit` function should be defined in the modal, since its the one with the submit button. Generally, React components should manage their own state as much as possible.